src/common/init.cpp

```diff
@@ -10,20 +10,24 @@
 //
 //===----------------------------------------------------------------------===//
 
-
 #include "common/init.h"
 
 #include "libcds/cds/init.h"
 
 #include <google/protobuf/stubs/common.h>
 
+#include <thread>
+
 namespace peloton {
 
+WorkerThreadPool thread_pool;
+
 void PelotonInit::Initialize() {
 
   // Initialize CDS library
   cds::Initialize();
 
+  thread_pool.Initialize(std::thread::hardware_concurrency());
```

Comment:
Yingjun preferred to make the thread pool global for anyone to use it wherever needed later on. Currently, only the server using it.